unit tests intermittently fail

```
ok      github.com/coreos/flannel/pkg/ip    0.005s  coverage: 15.6% of statements
ok      github.com/coreos/flannel/subnet    2.008s  coverage: 59.5% of statements
--- FAIL: TestRemote-32 (0.00 seconds)
    remote_test.go:72: GetNetworkConfig failed: Get http://127.0.0.1:9999/_/config: dial tcp 127.0.0.1:9999: connection refused
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x44b7c3]
goroutine 21 [running]:
runtime.panic(0x713b60, 0x918cb3)
    /usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
testing.func·006()
    /usr/local/go/src/pkg/testing/testing.go:416 +0x176
runtime.panic(0x713b60, 0x918cb3)
    /usr/local/go/src/pkg/runtime/panic.c:248 +0x18d
github.com/coreos/flannel/remote.doTestRemote(0x7fe6f451eab8, 0xc2080800c0, 0xc20807e090, 0x7606d0, 0xe)
    /home/travis/gopath/src/github.com/coreos/flannel/gopath/src/github.com/coreos/flannel/remote/remote_test.go:75 +0x883
github.com/coreos/flannel/remote.TestRemote(0xc20807e090)
    /home/travis/gopath/src/github.com/coreos/flannel/gopath/src/github.com/coreos/flannel/remote/remote_test.go:45 +0x2d5
testing.tRunner(0xc20807e090, 0x912ee0)
    /usr/local/go/src/pkg/testing/testing.go:422 +0x8b
created by testing.RunTests
    /usr/local/go/src/pkg/testing/testing.go:504 +0x8db
```

Same issue with the new PR. 

0K$ ./test
Building flanneld...
Running tests...
ok  	github.com/coreos/flannel/pkg/ip	0.005s	coverage: 15.6% of statements
ok  	github.com/coreos/flannel/subnet	2.009s	coverage: 59.5% of statements
--- FAIL: TestRemote-32 (0.00 seconds)
	remote_test.go:72: GetNetworkConfig failed: Get http://127.0.0.1:9999/_/config: dial tcp 127.0.0.1:9999: connection refused
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x44b7c3]

goroutine 21 [running]:
runtime.panic(0x713b60, 0x918cb3)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
testing.funcÂ·006()
	/usr/local/go/src/pkg/testing/testing.go:416 +0x176
runtime.panic(0x713b60, 0x918cb3)
	/usr/local/go/src/pkg/runtime/panic.c:248 +0x18d
github.com/coreos/flannel/remote.doTestRemote(0x7f33227e7b78, 0xc208088040, 0xc208086090, 0x7606d0, 0xe)
	/home/travis/gopath/src/github.com/coreos/flannel/gopath/src/github.com/coreos/flannel/remote/remote_test.go:75 +0x883
github.com/coreos/flannel/remote.TestRemote(0xc208086090)
	/home/travis/gopath/src/github.com/coreos/flannel/gopath/src/github.com/coreos/flannel/remote/remote_test.go:45 +0x2d5
testing.tRunner(0xc208086090, 0x912ee0)
	/usr/local/go/src/pkg/testing/testing.go:422 +0x8b
created by testing.RunTests
	/usr/local/go/src/pkg/testing/testing.go:504 +0x8db

goroutine 16 [chan receive]:
testing.RunTests(0x7f3e40, 0x912ee0, 0x1, 0x1, 0x6ee501)
	/usr/local/go/src/pkg/testing/testing.go:505 +0x923
testing.Main(0x7f3e40, 0x912ee0, 0x1, 0x1, 0x922a40, 0x0, 0x0, 0x922a40, 0x0, 0x0)
	/usr/local/go/src/pkg/testing/testing.go:435 +0x84
main.main()
	github.com/coreos/flannel/remote/_test/_testmain.go:97 +0x10c

--- a/remote/remote_test.go
+++ b/remote/remote_test.go
@@ -17,8 +17,11 @@ package remote
 import (
 	"fmt"
 	"net"
+	"net/url"
 	"sync"
+	"syscall"
 	"testing"
+	"time"
 
 	"github.com/coreos/flannel/Godeps/_workspace/src/golang.org/x/net/context"
 
@@ -64,16 +67,38 @@ func mustParseIP4Net(s string) ip.IP4Net {
 	return ip.FromIPNet(n)
 }
 
+func isConnRefused(err error) bool {
+	if uerr, ok := err.(*url.Error); ok {
+		if operr, ok := uerr.Err.(*net.OpError); ok {
+			return operr.Err == syscall.ECONNREFUSED
+		}
+	}
+	return false
+}
+
 func doTestRemote(ctx context.Context, t *testing.T, remoteAddr string) {
 	sm := NewRemoteManager(remoteAddr)
 
-	cfg, err := sm.GetNetworkConfig(ctx, "_")
-	if err != nil {
-		t.Errorf("GetNetworkConfig failed: %v", err)
-	}
+	for i := 0; ; i++ {
+		cfg, err := sm.GetNetworkConfig(ctx, "_")
+		if err != nil {
+			if isConnRefused(err) {
+				if i == 100 {
+					t.Fatalf("Out of connection retries")
+				}
 
-	if cfg.Network.String() != expectedNetwork {
-		t.Errorf("GetNetworkConfig returned bad network: %v vs %v", cfg.Network, expectedNetwork)
+				fmt.Println("Connection refused, retrying...")
+				time.Sleep(300 * time.Millisecond)
+				continue
+			}
+
+			t.Fatalf("GetNetworkConfig failed: %v", err)
+		}
+
+		if cfg.Network.String() != expectedNetwork {
+			t.Errorf("GetNetworkConfig returned bad network: %v vs %v", cfg.Network, expectedNetwork)
+		}
+		break
 	}
 
 	attrs := &subnet.LeaseAttrs{
@@ -81,7 +106,7 @@ func doTestRemote(ctx context.Context, t *testing.T, remoteAddr string) {
 	}
 	l, err := sm.AcquireLease(ctx, "_", attrs)
 	if err != nil {
-		t.Errorf("AcquireLease failed: %v", err)
+		t.Fatalf("AcquireLease failed: %v", err)
 	}
 
 	if !mustParseIP4Net(expectedNetwork).Contains(l.Subnet.IP) {

